malasakit-django/feature_phone/admin.py

```diff
@@ -16,6 +23,45 @@ class RecordingAdmin(admin.ModelAdmin):
     }
 
 
+    def download_list_action(modeladmin, request, queryset):
+
+
+      def zip_file(zf, obj, file_field, folder=False):
+        if hasattr(obj, file_field) and hasattr(getattr(obj, file_field),
+                                                'url'):
+          fpath = str(os.path.join(os.path.dirname(settings.MEDIA_ROOT),
+                                   str(getattr(obj, file_field).url)[1:]))
+          base = os.path.basename(fpath)
+          if folder:
+            base = file_field + '/' + base 
+          print(base)
+          zf.write(fpath, base)
+
+
+      zip_filename = 'audio.zip'
+      s = StringIO.StringIO()
+      with zipfile.ZipFile(s, 'w') as zf:
+        for obj in queryset:
+          zip_file(zf, obj, 'recording')
+          zip_file(zf, obj, 'age', folder=True)
+          zip_file(zf, obj, 'gender', folder=True)
+          zip_file(zf, obj, 'location', folder=True)
+
+      response = HttpResponse(s.getvalue(), content_type=
+                              'application/x-zip-compressed')
+      response['Content-Disposition'] = 'attachment; filename={}'.format(
+          zip_filename)
+      return response
+    download_list_action.short_description = 'Download selected mp3 files'
+
+
+    def download_button(self, obj):
+      return format_html('<a class="button" href="{}"download>Download</a>'
+                         .format(obj.recording.url))
+    download_button.short_description = 'mp3 file'
```

Comment:
Technically, this is not true--Twilio uses WAV files, not MP3. Since the admin panel supports both, just use "audio files" or "recordings".